src/main/scala/generate/database/PostgresGenerator.scala

```diff
@@ -6,17 +6,40 @@ import utils.StringUtils._
 /** Implementation of [[DatabaseGenerator]] for generating PostgreSQL */
 object PostgresGenerator extends DatabaseGenerator {
 
+  /** Given a comparison, parse it into the Postgres format */
+  private def parseComparison(comparison: Comparison): String =
+    comparison match {
+      case GreaterEqual => ">="
+      case Greater      => ">"
+      case Equal        => "=="
+      case NotEqual     => "!="
+      case Less         => "<"
+      case LessEqual    => "<="
+    }
+
+  /** Given a column constraint, parse it into the Postgres format */
+  private def parseConstraint(constraint: ColumnConstraint): String =
+    constraint match {
+      case NonNull                    => "NOT NULL"
+      case Null                       => "NULL"
+      case Check(left, comp, right)   => s"CHECK ($left " + parseComparison(comp) + s" $right)"
+      case Unique                     => "UNIQUE"
+      case PrimaryKey                 => "PRIMARY KEY"
+      case References(table, colName) => s"REFERENCES $table($colName)"
+    }
+
   /** Given a column, parse it into the type required by Postgres */
   private def parseColumn(column: Column): String =
     indent(
       column match {
-        case IntColumn(name)        => s"$name INT"
-        case FloatColumn(name)      => s"$name REAL"
-        case StringColumn(name)     => s"$name TEXT"
-        case BoolColumn(name)       => s"$name BOOLEAN"
-        case DateColumn(name)       => s"$name DATE"
-        case TimeColumn(name)       => s"$name TIME"
-        case DateTimeTzColumn(name) => s"$name TIMESTAMPTZ"
+        case IntColumn(name, constraints)    => s"$name INT" + constraints.map(parseConstraint).map(" " + _).mkString
```

Comment:
Maybe bring constraints up to the `trait` level, then perform the addition with constraints at the end?